client/src/cmdlf.c

```diff
@@ -689,109 +688,182 @@ int CmdLFConfig(const char *Cmd) {
 
     if (trigg > -1) {
         config.trigger_threshold = trigg;
-        gs_lf_threshold_set = (config.trigger_threshold > 0);
     }
 
     config.samples_to_skip = skip;
     return lf_config(&config);
 }
 
-int lf_read(bool verbose, uint32_t samples) {
+static int lf_read_internal(bool realtime, bool verbose, uint64_t samples) {
     if (!g_session.pm3_present) return PM3_ENOTTY;
 
-    struct p {
-        uint32_t samples : 31;
-        bool     verbose : 1;
-    } PACKED;
-
-    struct p payload;
+    lf_sample_payload_t payload;
```

Comment:
Could this leave fields of the struct uninitialized?
Even if not true now, still recommend to zero the full `sizeof()` a struct when it is allocated ... else padding bits can be in undefined states, which is technically fine, but annoying when debugging.   Also, longer-term, if the struct changes and an individual use isn't updated, zero'ing the full struct ensures at least repeatable (if not correct) behavior, simplifying tracking bugs down.

<details><summary>Contrast with ... </summary><p/>

Contrast with `sample_config` below, which **_immediately_** calls `lf_getconfig()`, which ensures the struct is fully initialized on success ... that style of constructor-like behavior is likely an allowed exception.

</details>

Reply:
I think `lf_sample_payload_t payload = {0};` is good for it, right?

Reply:
Fixed